File: src/API/Arbitragem.Application/ViewModels/OpportunityViewModel.cs
namespace Arbitragem.Application.ViewModels;

public record OpportunityViewModel(
    int Position,
    double ValueToBuy,
    double ValueToSell,
    double DifferencePercentage,
    Guid CryptoId,
    string Symbol,
    string Name,
    Guid ExchangeToBuyId,
    string ExchangeToBuyName,
    Guid ExchangeToSellId,
    string ExchangeToSellName,
    DateTime Date,
    string ExchangeToBuyUrl,
    string ExchangeToSellUrl,
    double Fee
    );

File: src/ArbitraX/src/app/core/ViewModels/OpportunityViewModel.ts
export interface OpportunityViewModel {
  position: number;
  valueToBuy: number;
  valueToSell: number;
  differencePercentage: number;
  cryptoId: string;
  symbol: string;
  name: string;
  exchangeToBuyId: string;
  exchangeToBuyName: string;
  exchangeToSellId: string;
  exchangeToSellName: string;
  date: Date;
  exchangeToBuyUrl: string;
  exchangeToSellUrl: string;
}

export interface GetAllResponse {
  opportunities: OpportunityViewModel[];
  dolar: number;
}
